app/controllers/users_controller.rb

```diff
@@ -147,6 +141,28 @@ def reset_user_password
     redirect_to users_path
   end
 
+  def emails_csv
+    @teams = Team.all
+    @rps, @idps, @other = @teams.group_by(&:team_type).values_at(TEAMS::RP, TEAMS::IDP, TEAMS::OTHER)
+    @all_rps_and_idps = @teams - @other unless @other.nil?
+
+    if params['team'] == TEAMS::RP
+      @emails = helpers.get_all_emails_for_users_in_teams(@rps)
+    elsif params['team'] == TEAMS::IDP
+      @emails = helpers.get_all_emails_for_users_in_teams(@idps)
+    elsif params['team'] == TEAMS::ALL
+      @emails = helpers.get_all_emails_for_users_in_teams(@all_rps_and_idps)
```

Comment:
Not sure about the location of `get_all_emails_for_users_in_teams` in AdminHelper maybe we need a new Team oriented helper and these methods can go in there
`get_all_emails_for_users_in_teams`
`idp_team_user?`
`rp_team_user?`
`get_team_members_from_group(team) #get_users_in_group(group_name: @team.team_alias).map { |cognito_user| as_team_member(cognito_user: cognito_user) }`
`team_recipients`
`as_team_member` maybe?

Reply:
This admin helper method `get_team_members_from_group` can actually be moved into the controller itself as a private method. The other methods are used in the view so I guess they can stay there. I also removed the rp_team_user method as it was not going to be used. So technically after I push up my change theres only one method to determine if its an idp_team in there